### PR TITLE
Add -with-annotation flag to hcl2_upgrade command

### PR DESCRIPTION
**Pull request by sylviamoss**
This PR adds the `-with-annotations` to make helper annotations optional and not the default. Besides that, this PR also refactors a little bit the code.

### CODE REVIEW

**Review by SwampDragons — APPROVED** [review]
Looks good but can you also add the flag to the "help" text?

``` 
mmarsh@MacBook-Pro  ~/dev/repro_cases  PACKER_LOG=0 packer hcl2_upgrade --help
Usage: packer hcl2_upgrade -output-file=JSON_TEMPLATE.pkr.hcl JSON_TEMPLATE...

  Will transform your JSON template into an HCL2 configuration.
```

**Comment by SwampDragons** [comment]
Ah, that's a silly thing to hold this on-- Merging, and I'll make a separate PR with the help text

File: command/hcl2_upgrade.go [hunk 1]
```diff
@@ -221,248 +192,71 @@ func (c *HCL2UpgradeCommand) RunContext(_ context.Context, cla *HCL2UpgradeArgs)
...
-
-	if len(amazonSecretsManagerMap) > 0 {
-		out.Write([]byte(amazonSecretsManagerDataHeader))
-		out.Write(amazonSecretsDataOut)
-	}
-
-	if len(amazonAmiOut) > 0 {
-		out.Write([]byte(amazonAmiDataHeader))
-		out.Write(amazonAmiOut)
+	blocks := map[int]BlockParser{
```
Comment (azr) [inline command/hcl2_upgrade.go:238]: Why not simply using a slice here ?

File: command/hcl2_upgrade.go [hunk 2]
```diff
@@ -221,248 +192,71 @@ func (c *HCL2UpgradeCommand) RunContext(_ context.Context, cla *HCL2UpgradeArgs)
...
+		6: sources,
+		7: build,
 	}
-
-	if timestamp {
-		_, _ = out.Write([]byte("\n"))
-		fmt.Fprintln(out, `# "timestamp" template function replacement`)
-		fmt.Fprintln(out, `locals { timestamp = regex_replace(timestamp(), "[- TZ:]", "") }`)
+	for i := 1; i <= len(blocks); i++ {
+		blocks[i].Write(out)
```
Comment (azr) [inline command/hcl2_upgrade.go:248]: nice !!

File: command/hcl2_upgrade.go (hunk 1 above)
Comment (sylviamoss) [reply command/hcl2_upgrade.go:238]: You're right, I forgot at the time that slices are ordered by append 🤦🏼

**Review by azr — COMMENTED** [review]
nice one !

**Pull request merged**